### DISCUSSION TITLE
The repository 'https://ppa.launchpadcontent.net/nginx/stable/ubuntu noble Release' does not have a Release file

### DISCUSSION THREAD

**Discussion by mjau-mjau** [oerdnj/deb.sury.org#2206] [Q&A] [answered]
Hi. After installing `ppa:ondrej/php`, I get the following error whenever trying to update `apt update` on Ubuntu `24.04`:

```
The repository 'https://ppa.launchpadcontent.net/nginx/stable/ubuntu noble Release' does not have a Release file
```

In full context:
```
:~# sudo apt update
Hit:1 http://archive.ubuntu.com/ubuntu noble InRelease
Hit:2 http://ubuntu.mirror.constant.com noble InRelease
Hit:3 http://archive.ubuntu.com/ubuntu noble-updates InRelease
Hit:4 http://security.ubuntu.com/ubuntu noble-security InRelease
Hit:5 http://ubuntu.mirror.constant.com noble-updates InRelease
Hit:6 http://archive.ubuntu.com/ubuntu noble-backports InRelease
Hit:7 http://archive.ubuntu.com/ubuntu noble-security InRelease
Hit:8 http://ubuntu.mirror.constant.com noble-backports InRelease
Hit:9 http://ubuntu.mirror.constant.com noble-security InRelease
Hit:10 http://us.archive.ubuntu.com/ubuntu noble InRelease
Ign:11 https://ppa.launchpadcontent.net/nginx/stable/ubuntu noble InRelease
Hit:12 http://us.archive.ubuntu.com/ubuntu noble-updates InRelease
Hit:13 https://ppa.launchpadcontent.net/ondrej/php/ubuntu noble InRelease
Hit:14 http://us.archive.ubuntu.com/ubuntu noble-backports InRelease
Err:15 https://ppa.launchpadcontent.net/nginx/stable/ubuntu noble Release
  404  Not Found [IP: 2620:2d:4000:1::81 443]
Reading package lists... Done
E: The repository 'https://ppa.launchpadcontent.net/nginx/stable/ubuntu noble Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

The problematic file seems to be `/etc/apt/sources.list.d/nginx-ubuntu-stable-noble.sources`, which got added after install. If I temporarily disable the file, the update can proceed as normal. Does anyone have an explanation or solution for this?

**Comment by oerdnj** [comment]
Who added the `/etc/apt/sources.list.d/nginx-ubuntu-stable-noble.sources` file? It didn't magically appear.

And what are the contents of the file?

**Reply by mjau-mjau** [reply]
> Who added the `/etc/apt/sources.list.d/nginx-ubuntu-stable-noble.sources` file? It didn't magically appear.

It was added or modified at the same time as installing `ppa:ondrej/php`.

![](https://i.imgur.com/EYOPFdr.jpeg)

> And what are the contents of the file?

```
Types: deb
URIs: https://ppa.launchpadcontent.net/nginx/stable/ubuntu/
Suites: noble
Components: main
Signed-By: 
 -----BEGIN PGP PUBLIC KEY BLOCK-----
 {SOMEKEYBEHERE}
 -----END PGP PUBLIC KEY BLOCK-----
```

**Comment by oerdnj** [comment, marked as answer]
> It was added or modified at the same time as installing ppa:ondrej/php.

Nope, that's not true. This repository has nothing to do with ppa:ondrej/php. Check your scripts.

**Reply by mjau-mjau** [reply]
Sorry, it was caused by `add-apt-repository ppa:nginx/stable` recommended by various Nginx upgrade [tutorials](https://medium.com/@mustafaaltunok/a-step-by-step-guide-to-upgrade-nginx-to-the-latest-stable-version-on-ubuntu-cbfac70afb65), yet the repository has expired.
